towhee/hub/hub_tools.py

```diff
@@ -385,6 +388,126 @@ def download_files(user: str, repo: str, branch: str, file_list: List[str], lfs_
+
+    Args:
+        user (`str`):
+            The account name.
+        repo (`str`):
+            The repo name.
+        repo_type (`str`):
+            Which category of repo to create, only one can be used, includes
+            ('model', 'operator', 'pipeline', 'dataset').
+
+    Raises:
+        (`HTTPError`)
+            Raise error in request.
+        (`OSError`)
+            Raise error in writing file.
+    """
+    links = 'https://hub.towhee.io/' + user + '/' + repo + '.git'
+    subprocess.call(['git', 'clone', links])
+    repo_file_name = repo.replace('-', '_')
+    if repo_type == 'operator':
+        download_repo('towhee', 'operator-template', 'main', os.getcwd() + '/' + repo)
+        os.rename(repo + '/operator_template.py', repo + '/' + repo_file_name + '.py')
+        os.rename(repo + '/operator_template.yaml', repo + '/' + repo_file_name + '.yaml')
+    elif repo_type == 'pipeline':
+        download_repo('towhee', 'pipeline-template', 'main', os.getcwd() + '/' + repo)
+        os.rename(repo + '/pipeline_template.yaml', repo + '/' + repo_file_name + '.yaml')
+
+
+def covert_dic(dicts: dict) -> dict:
+    """
+    Convert all the values in a dictionary to str and replace char, for example:
+     <class 'torch.Tensor'>(unknow type) to torch.Tensor(str type).
+
+    Args:
+        dicts (`dict`):
+            The dictionary to convert.
+
+    Returns:
+        (`dict`)
+            The converted dictionary.
+    """
+    for keys in dicts:
+        dicts[keys] = str(dicts[keys]).replace('<class ', '').replace('>', '').replace('\'', '')
+    return dicts
+
+
+def generate_repo_yaml(user: str, repo: str) -> None:
+    """
+    Generate the yaml of Operator, for example:
+    name: 'operator-template'
+    labels:
+      recommended_framework: ''
+      class: ''
+      others: ''
+    operator: 'towhee/operator-template'
+    init:
+      model_name: str
+    call:
+      input:
+        img_path: str
+      output:
+        feature_vector: numpy.ndarray
+
+    Args:
+        user (`str`):
+            The account name.
+        repo (`str`):
+            The repo name.
+
+    Raises:
+        (`HTTPError`)
+            Raise error in request.
+        (`OSError`)
+            Raise error in writing file.
+    """
+    sys.path.append(repo)
+    repo_file_name = repo.replace('-', '_')
+    # get class name in camel case
+    components = repo.split('-')
+    class_name = ''.join(x.title() for x in components)
+    yaml_file = repo + '/' + repo_file_name + '.yaml'
+    operator_name = user + '/' + repo
+    # import the class from repo
+    cls = getattr(import_module(repo_file_name, repo), class_name)
+
+    init_args = cls.__init__.__annotations__
+    try:
+        del init_args['return']
+    except KeyError:
+        pass
+
+    call_func = cls.__call__.__annotations__
+    try:
+        call_output = call_func.pop('return')
+        call_output = call_output._field_types
```

Comment:
Directly accessing a field beginning with an underscore will cause `pylint` to fail.

Reply:
Actually the `_field_defaults` is [the attribute of a Namedtuple](https://pythontic.com/containers/namedtuple/_field_defaults), for example, we can get the Outputs info with it:
```python
>> from resnet_image_embedding import ResnetImageEmbedding
>>> call = ResnetImageEmbedding.__call__.__annotations__
>>> call
{'img_tensor': <class 'torch.Tensor'>, 'return': <class 'resnet_image_embedding.Outputs'>}
>>> re = call['return']
>>> re
<class 'resnet_image_embedding.Outputs'>
>>> re._field_types
{'feature_vector': <class 'numpy.ndarray'>}
```